pkg/models/node_info.go

```diff
@@ -77,7 +85,12 @@ type NodeInfo struct {
 
 // ID returns the node ID
 func (n NodeInfo) ID() string {
-	return n.PeerInfo.ID.String()
+	if n.NodeID != "" {
+		return n.NodeID
+	} else if n.PeerInfo != nil {
+		return n.PeerInfo.ID.String()
+	}
+	return ""
```

Comment:
Doesn't seem like "" would be valid.

Reply:
it doesn't, but I prefer that `ID()` method doesn't return errors as well.  Maybe we just need to make `NodeInfo` implement `Validate` method and make sure it is called whenever it is instantiated. Added a TODO in https://github.com/bacalhau-project/bacalhau/pull/3188/commits/8ffa1caecd1f42d866d80a27e8b4adf292bf7eca